envolve-backend/src/main/java/de/neuefische/erbay/envolve/service/UserService.java

```diff
@@ -0,0 +1,24 @@
+package de.neuefische.erbay.envolve.service;
+
+import de.neuefische.erbay.envolve.db.TeacherDb;
+import de.neuefische.erbay.envolve.model.Teacher;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserService {
+    private final TeacherDb teacherDb;
+    BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
```

Comment:
you can use inject PasswordEncoder  instead of hard code your encoder.

Then you get the encoder form the config